### ISSUE TITLE
0.0 / 0.0 is computed inconsistently across OSes

### ISSUE THREAD

**Issue by HansKristian-ARM** [KhronosGroup/glslang#1552]
This shader: https://github.com/KhronosGroup/SPIRV-Cross/blob/bitcast-builtin-compute/shaders/frag/inf-nan-constant.frag generates a NaN, but the bit pattern in SPIR-V seems to depend on OS. On Linux and OSX, the sign bit is not set, but on Windows, it is set, causing test issues since the output SPIR-V is not the same.

**Comment by johnkslang** [comment]
Is this supposed to be supported? It seems a bit like a layered interpretation that is not guaranteed to work, followed by surprise it is not portable, but I haven't gone and done the spec. archaeology yet.

E.g., there is a reason we have `isnan(x)` rather than `if (x == 0.0/0.0)`, and that SPIR-V says:

> The resulting value is undefined if Operand 2 is 0.

(Though, I know you are not trying to divide on the target.)

We also say in GLSL:

> Non-specialization constant expressions may be evaluated by the compiler’s host platform, and are
therefore not required to compute the same value that the same expression would evaluate to on
the shader execution target.

**Comment by HansKristian-ARM** [comment]
Right, I'm not sure if this should be consistent according to spec, but it is a bit surprising that the same code on same compiler (but different OS) generates a different result, just want to make sure this is not intentional, because it's certainly surprising.

The background for this was simply a test in SPIRV-Cross to make sure I could deal with Inf and NaN OpConstants. I've since moved the test to asm.

**Comment by johnkslang** [comment]
The spec does not pin the answer down, because different hardware does give different results. If you want an IEEE NaN or INF, I suggest you just directly use the bit pattern you want, and not try to make the host create one with non-portable expressions.
